dotCMS/src/main/java/com/dotcms/rendering/velocity/services/VTLLoader.java

```diff
@@ -0,0 +1,144 @@
+package com.dotcms.rendering.velocity.services;
+
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.ConfigUtils;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+
+import org.apache.velocity.exception.ResourceNotFoundException;
+
+import com.liferay.util.FileUtil;
+
+public class VTLLoader implements DotLoader {
+    private static String velocityCanoncalPath;
+    private static String assetCanoncalPath;
+    private static String assetRealCanoncalPath;
+    private static String VELOCITY_ROOT;
+    private static VTLLoader vtlServices;
+
+    private VTLLoader() {
+        init();
+    }
+
+    public static VTLLoader instance() {
+        if (vtlServices == null) {
+            synchronized (VTLLoader.class) {
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Remove this dangerous instance of double-checked locking. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2168)